addons/account_update_tax_tags/tests/test_account_update_tax_tags_wizard.py

```diff
@@ -0,0 +1,325 @@
+
+    def test_update_move_type_entry(self):
+        """ Test that move of type 'entry' are correctly updated.
+        If a move has negative balance and use a sale tax, it should act as an invoice.
+        If a move has positive balance and use a sale tax, it should act as a refund.
+        If a move has negative balance and use a purchase tax, it should act as an invoice.
+        If a move has positive balance and use a purchase tax, it should be treated as an invoice.
+        """
+        account = self.company_data['default_account_assets']  # Account is not relevant for the test but must be set.
+        for tax_type in ['sale', 'purchase']:
+            tax = self._create_tax(f'test_{tax_type}_tax', amount=10, type_tax_use=tax_type, tag_names=self.tag_names)
+            for balance in [-1000, 1000]:
+                with self.subTest(f'Testing move type entry {tax_type}-{balance}'):
+
+                    # Create one entry and it's reverse.
+                    move_form = Form(self.env['account.move'].with_context(default_move_type='entry'))
+                    with move_form.line_ids.new() as line:
+                        line.account_id = account
+                        if balance < 0:
+                            line.credit = abs(balance)
+                        else:
+                            line.debit = abs(balance)
+                        line.tax_ids.add(tax)
+
+                    # Create a third account.move.line for balance. It will be automatically filled given above lines.
+                    move_form.line_ids.new().save()
+                    move = move_form.save()
+
+                    self._change_tax_tag(tax, 'test_invoice_tag_changed', invoice=True, base=True)
+                    self._change_tax_tag(tax, 'test_refund_tag_changed', invoice=False, base=True)
+
+                    self.wizard.update_amls_tax_tags()
+                    if (balance < 0 and tax_type == 'sale') or (balance > 0 and tax_type == 'purchase'):
+                        self.assertEqual(move.invoice_line_ids.tax_tag_ids.name, 'test_invoice_tag_changed')
+                    elif (balance < 0 and tax_type == 'purchase') or (balance > 0 and tax_type == 'sale'):
+                        self.assertEqual(move.invoice_line_ids.tax_tag_ids.name, 'test_refund_tag_changed')
+
+    def test_update_amls_all_states(self):
+        """ Tests that moves are correctly updated, regardless of their state. """
+        self._change_tax_tag(self.tax_1, 'invoice_tax_tag_changed', invoice=True, base=False)
+        for state in ['posted', 'cancel', 'draft']:
+            with self.subTest(f'Update tax tag on move with state: {state}'):
+                move = self._create_move(partner_id=self.partner_a.id, tax_ids=self.tax_1)
+                if state == 'cancel':
+                    move.write({'state': 'cancel'})
+                elif state == 'posted':
+                    move.action_post()
```

Comment:
This could be part of the `_create_move` api and making it usable like this:
```py
move = self._create_move(partner_id=self.partner_a.id, tax_ids=self.tax_1, state=state)
```